### PR TITLE
mypy typing issues for decimal pandas/tests/extension/decimal/test_decimal.py

### PR DESCRIPTION
**Pull request by yogendrasoni**
- [ ] xref #28926
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -155,7 +155,7 @@ ignore_errors=True
 ignore_errors=True
 
 [mypy-pandas.tests.extension.decimal.test_decimal]
-ignore_errors=True
+ignore_errors=False
```
Comment (simonjayhawkins) [inline setup.cfg]: can you remove this entire section from setup.cfg

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@yogendrasoni Thanks for the PR. A couple of suggestions.

File: pandas/tests/extension/decimal/test_decimal.py [hunk 2]
```diff
@@ -110,35 +110,35 @@ def assert_frame_equal(self, left, right, *args, **kwargs):
         tm.assert_frame_equal(left, right, *args, **kwargs)
 
 
-class TestDtype(BaseDecimal, base.BaseDtypeTests):
+class TestDtype(BaseDecimal, base.BaseDtypeTests):  # type: ignore
```
[the remark this answers is not in the document]
Comment (yogendrasoni) [reply pandas/tests/extension/decimal/test_decimal.py]: tried putting staticmethod, BaseExtensionTests methods takes (left,right, some arguments) but BaseDecimal takes (left,right,*args,**kwargs) it is still showing incompatible by mypy, didn't find anything to solve this. I need to study more to solve this.

[the remark this answers is not in the document]
Comment (WillAyd) [reply pandas/tests/extension/decimal/test_decimal.py]: I think #28994 has the same issue. What error are you getting after trying what @simonjayhawkins suggested?

**Comment by jbrockmendel** [comment]
@simonjayhawkins am i right in thinking that the `# type: ignore` this uses is something were trying pretty hard to avoid?

File: pandas/tests/extension/decimal/test_decimal.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/tests/extension/decimal/test_decimal.py]: yes we have two modules that are reporting mypy errors due to the same root issue.

**Comment by simonjayhawkins** [comment]
> @simonjayhawkins am i right in thinking that the `# type: ignore` this uses is something were trying pretty hard to avoid?

correct.

**Comment by WillAyd** [comment]
@yogendrasoni is this still active?

**Comment by jreback** [comment]
closing as stale, if you'ld like to continue, pls ping.

**Pull request closed without merging**